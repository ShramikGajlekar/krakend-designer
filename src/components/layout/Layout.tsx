import { makeStyles } from '@material-ui/core';
import Container from '@material-ui/core/Container';
import React, { FunctionComponent } from 'react';
import { Header, HeaderProps } from '../header/Header';

const useStyles = makeStyles({
    container: {
        margin: '0',
        padding: '0',
    },
});

export interface LayoutProps {
    headerOptions: HeaderProps;
}

export const Layout: FunctionComponent<LayoutProps> = ({ headerOptions, children }) => {
    const classes = useStyles();
    return (
        <>
            <Container maxWidth={false} className={classes.container}>
                <Header title={headerOptions.title} logoURL={headerOptions.logoURL} />
                {children}
            </Container>
        </>
    );
};
