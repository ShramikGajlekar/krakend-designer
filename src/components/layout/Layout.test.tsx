import { render } from '@testing-library/react';
import React from 'react';
import { LayoutProps, Layout } from './Layout';

describe('Layout with props', () => {
    it('match a snapshot', () => {
        const props: LayoutProps = {
            headerOptions: {
                title: 'title',
            },
        };
        const { container } = render(<Layout headerOptions={props.headerOptions} />);
        expect(container).toMatchSnapshot();
    });
});
