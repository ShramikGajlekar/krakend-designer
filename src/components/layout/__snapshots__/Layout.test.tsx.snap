// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout with props match a snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-container-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-3 mui-fixed MuiPaper-elevation4"
      style="margin: 0px;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          title
        </div>
        <div
          style="margin-left: auto;"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Tab1
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Tab2
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Tab3
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</div>
`;
